### PR TITLE
test: added test case function for merge sort

### PR DESCRIPTION
**Pull request by Nishant2209**
test: added test case feature for the merge sort algorithm
feat: added 2 operation modes for the user

#### Description of Change
The user can operate in two modes
 1) Interactive Mode - in which user can enter their custom values in an array and sort that array. 
 2) Automatic Mode - in which the pre-generated test cases will run (this feature was not present before).

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by Nishant2209** [comment]
test: increase the number of test cases

File: sorting/merge_sort.cpp [hunk 1]
```diff
@@ -30,7 +31,11 @@
  *
  * @param r - end index or right index of second half array
  */
-void merge(int *arr, int l, int m, int r) {
+
+namespace sorting {
+namespace mergesort {
```
Comment (Panquesito7) [inline sorting/merge_sort.cpp:36]:
```suggestion
/**
 * @namespace sorting
 * @brief Sorting algorithms
 */
namespace sorting {
/**
 * @namespace mergesort
 * @brief Functions for the [Merge Sort](https://en.wikipedia.org/wiki/Merge_sort) algorithm implementation
 */
namespace mergesort {
```

File: sorting/merge_sort.cpp [hunk 2]
```diff
@@ -86,20 +93,83 @@ void show(int *arr, int size) {
...
+    sorting::mergesort::mergeSort(arr4, 0, size4 - 1);
+    std::cout << "Passed\n";
+    sorting::mergesort::mergeSort(arr4, 0, size4 - 1);
+    for (int i = 0; i < size4; i++) {
+        std::cout << arr4[i] << ", ";
+    }
+    std::cout << std::endl;
+}
+
 /** Main function */
```
Comment (Panquesito7) [inline sorting/merge_sort.cpp:150]:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
```

**Review by Panquesito7 — COMMENTED** [review]
Great work! 🚀

**Pull request closed without merging**